app/src/features/apiClient/screens/apiClient/utils.ts

```diff
@@ -575,3 +581,30 @@ export const processRecordsForDuplication = (recordsToProcess: RQAPI.Record[]) =
 
   return recordsToDuplicate;
 };
+
+export const getUpdatedVariableCount = (originalData: EnvironmentVariables, updatedData: EnvironmentVariables) => {
+  const formattedData = Object.entries(originalData).map(([key, value]) => ({
+    key,
+    type: value.type,
+    localValue: value.localValue,
+    syncValue: value.syncValue,
+    id: value.id,
+  }));
+
+  const formattedChangedData = Object.entries(updatedData).map(([key, value]) => ({
+    key,
+    type: value.type,
+    localValue: value.localValue,
+    syncValue: value.syncValue,
+    id: value.id,
+  }));
+
+  return differenceWith(formattedChangedData, formattedData, isEqual).length;
+};
+
+export const trackUserProperties = (records: RQAPI.Record[]) => {
+  const totalCollections = records.filter((record) => isApiCollection(record)).length;
+  const totalRequests = records.length - totalCollections;
+  submitAttrUtil(APP_CONSTANTS.GA_EVENTS.ATTR.NUM_COLLECTIONS, totalCollections);
+  submitAttrUtil(APP_CONSTANTS.GA_EVENTS.ATTR.NUM_REQUESTS, totalRequests);
+};
```

Comment:
Can we not have this here in common utils? Can we please move it to analytics folder or the local to the file where its being used?